multigeiger/transmission.cpp

```diff
@@ -296,5 +311,59 @@ void transmit_data(String tube_type, int tube_nbr, unsigned int dt, unsigned int
     set_status(STATUS_TTN, ttn_ok ? ST_TTN_IDLE : ST_TTN_ERROR);
     display_status();
   }
+
+}
+
+void transmit_userinfo(String tube_type, int tube_nbr, float tube_factor, unsigned int cpm, unsigned int accu_cpm, float accu_rate,
+                   int have_thp, float temperature, float humidity, float pressure, int wifi_status, bool alarm_status) {
+
+  if (wifi_status != ST_WIFI_CONNECTED)
+    return;
+
+  // if we don't have a valid Messenger config, do not send to Messenger
+  if (sendDataToMessengerEvery < 0)
+    return;
+  // in case the config has been changed in the meantime
+  if ((strlen(telegramBotToken) < 40) || (strlen(telegramChatId) < 7)) {
+    sendDataToMessengerEvery = -1;
+    return;
+  }
+
+  static unsigned int transmitCounter = 0;
+  transmitCounter++;
+  if (transmitCounter == 27720) transmitCounter = 0;  // 27720 % 1..12 == 0
+
+  char localEspId[16];
+  char thp_text[60];
+
+  chipID.toCharArray(localEspId, sizeof(localEspId));
+  if (have_thp) {
+    sprintf(thp_text, "\nBME data: %.1fC %.1f%% %.1fhPa", temperature, humidity, pressure/100);
+  }
+
+  if (((sendDataToMessengerEvery > 0) && (transmitCounter % sendDataToMessengerEvery == 0))
+      || ((sendDataToMessengerEvery >= 0) && alarm_status)) {
```

Comment:
this seems to be a triple use variable / in band signalling and that's bad:
- it is the interval, if > 0
- it means only send when alarm is on when 0
- it means disabled when < 0

can we have different variables to make it better readable / configurable?

e.g. could be one for the interval and another one for the mode (disabled, always, only on alarm).

Reply:
that would maybe also solve the use case "send in intervals of N" but "only when alarm is on".